config/dynamic.c

```diff
@@ -16,11 +16,529 @@
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to call setsockopt(SO_RCVBUF) with size = %d - (%s)\n",
+			rcv_buff_size, strerror(errno));
+		goto close_fd;
+	}
+
+	lua_state = luaL_newstate();
+	if (lua_state == NULL) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: failed to create new Lua state!\n");
+		goto close_fd;
+	}
+
+	/* Set up the Lua state while there is a connection. */
+	ret = setup_dy_lua(lua_state);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to set up the lua state\n");
+		goto close_lua;
+	}
+
+	while (1) {
+		ret = process_single_cmd(conn_fd, lua_state);
+		if (ret != 0)
+			break;
+	}
+
+close_lua:
+	lua_close(lua_state);
+
+close_fd:
+	ret = close(conn_fd);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to close the connection socket - (%s)\n",
+			strerror(errno));
+	}
+}
+
+static int
+dyn_cfg_proc(void *arg)
+{
+	int ret = 0;
+	uint32_t lcore = rte_lcore_id();
+	struct dynamic_config *dy_conf = arg;
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block is running at lcore = %u\n", lcore);
+
+	while (likely(!exiting)) {
+		fd_set fds;
+		struct timeval stv;
+ 
+		FD_ZERO(&fds);
+		FD_SET(dy_conf->sock_fd, &fds);
+ 
+		/*
+		 * 10000 usecs' timeout for the select() function.
+		 * This parameter can prevent the select() function
+		 * from blocking forever. So, the whole program can
+		 * exit when receiving a quitting signal.
+		 */
+		stv.tv_sec = 0;
+		stv.tv_usec = 10000;
+
+		ret = select(dy_conf->sock_fd + 1, &fds, NULL, NULL, &stv);
+		if (ret < 0 && errno != EINTR) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"dyn_cfg: Failed to call the select() function - (%s)\n",
+				strerror(errno));
+			break;
+		} else if (likely(ret <= 0))
+			continue;
+
+		/*
+		 * The config component accepts only one connection at a time.
+		 */
+		RTE_VERIFY(FD_ISSET(dy_conf->sock_fd, &fds));
+		handle_client(dy_conf->sock_fd, dy_conf);
+	}
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block at lcore = %u is exiting\n", lcore);
+
+	cleanup_dy(dy_conf);
+
+	return ret;
+}
+
+struct dynamic_config *
+get_dy_conf(void)
+{
+	return &config;
+}
+
+void
+set_dyc_timeout(unsigned int sec,
+	unsigned int usec, struct dynamic_config *dy_conf)
+{
+	dy_conf->rcv_time_out.tv_sec = sec;
+	dy_conf->rcv_time_out.tv_usec = usec;
+}
 
 int
-run_dynamic_config(__attribute__((unused)) const struct dynamic_config *dy_conf)
+run_dynamic_config(const char *server_path, struct dynamic_config *dy_conf)
 {
-	/* TODO Initialize and run Dynamic Config functional block. */
+	int ret;
+	struct sockaddr_un server_addr;
+
+	if (server_path == NULL || dy_conf == NULL) {
+		ret = -1;
+		goto out;
+	}
+
+	dy_conf->sock_fd = -1;
+
+	dy_conf->server_path = rte_malloc(
+		"server_path", strlen(server_path) + 1, 0);
+	if (dy_conf->server_path == NULL) {
+		ret = -1;
+		goto out;
+	}
+	strcpy(dy_conf->server_path, server_path);
+
+	/*
+	 * Remove any old socket and create an unnamed socket for the server.
+	 */
```

Comment:
If we put this comment into a single line, it will exceed 80 characters, so I separated them into three lines.